presentation/chart_view.py

```diff
@@ -0,0 +1,63 @@
+import urllib.parse as urlparse
+from datetime import datetime, timedelta, timezone
+from typing import List, Tuple
+from urllib.parse import parse_qs, unquote
+
+import dash_core_components as dcc
+import dash_html_components as html
+import plotly.graph_objs as go
+
+from domain.agents import Agents
+from domain.metric import Metric
+from domain.model import MeshResults
+from domain.model.mesh_results import HealthItem, MeshResults
+from domain.types import AgentID
+from presentation.health import Health
+
+
+class ChartView:
+    @staticmethod
+    def make_layout(from_agent, to_agent: AgentID, metric: Metric, results: MeshResults) -> html.Div:
+        # make chart title
+        agents = Agents(results)
+        from_alias = agents.get_alias(from_agent)
+        to_alias = agents.get_alias(to_agent)
+        title = f"{metric.value}: {from_alias} -> {to_alias}"
+
+        # make chart data
+        health = Health(from_agent, to_agent, results)
```

Comment:
Filtering of results to a particular agent pair and metric should be a method of the `MeshResults` class.

Reply:
Not sure if data filtering and transformation should be part of the data structure itself; you can come up with different filters and transformers and this usually only bloats the class. I dont see profit here

Reply:
After discussion, moving filtering to MeshResults